frontend/views/pages/Join.vue

```diff
@@ -1,79 +1,153 @@
+      svg-broken-link.c-svg
+      i18n.title.is-1(
+        v-if='isStatus("EXPIRED")'
+        tag='h1'
+        :args='{ ...LTags() }'
+      ) Oh no! {br_} Something went wrong.
+      i18n.title.is-1(
+        v-else
+        tag='h1'
+        :args='{ ...LTags() }'
+      ) Oh no! {br_} Your link has expired.
+      p.has-text-1 {{ ephemeral.errorMsg }}
+      i18n.c-goHome(tag='button' @click='goHome') Take me home
+
+  // TODO - Remove these. Only used for debug.
+  .buttons.c-debug
+    button.is-small.is-outlined(@click='setStatus("LOADING")') LOADING
+    button.is-small.is-outlined(@click='setStatus("SIGNING")') SIGNING
+    button.is-small.is-outlined(@click='setStatus("WELCOME")') WELCOME
+    button.is-small.is-outlined(@click='setStatus("EXPIRED")') EXPIRED
+    button.is-small.is-outlined(@click='setStatus("INVALID")') INVALID
 </template>
 
 <script>
 import sbp from '~/shared/sbp.js'
-import L from '@view-utils/translations.js'
-import Page from './Page.vue'
-import Bars from '@components/Graphs/Bars.vue'
+import { mapGetters } from 'vuex'
+import FormSignup from '@containers/forms/FormSignup.vue'
+import FormLogin from '@containers/forms/FormLogin.vue'
 import Loading from '@components/Loading.vue'
-import MembersCircle from '@components/MembersCircle.vue'
+import Avatar from '@components/Avatar.vue'
+import SvgBrokenLink from '@svgs/broken-link.svg'
+import L from '@view-utils/translations.js'
+import GroupWelcome from '@components/GroupWelcome.vue'
 
 export default {
   name: 'Join',
   components: {
-    Bars,
     Loading,
-    MembersCircle,
-    Page
+    FormLogin,
+    FormSignup,
+    Avatar,
+    GroupWelcome,
+    SvgBrokenLink
   },
   data () {
     return {
       ephemeral: {
-        contract: { members: [], settings: {} },
-        errorMsg: null
+        pageStatus: 'LOADING', // || WELCOME || SIGNING || LOGGING || EXPIRED || INVALID
+        invitation: {}
       }
     }
   },
+  computed: {
+    ...mapGetters(['ourUsername'])
+  },
   async mounted () {
     try {
-      const state = await sbp('state/latestContractState', this.$route.query.groupId)
-      // TODO: use the state.profiles directly?
-      var members = []
-      for (const name of Object.keys(state.profiles)) {
-        members.push(await sbp('state/latestContractState', state.profiles[name].contractID))
+      if (this.ourUsername) {
+        if (this.$store.state.contracts[this.$route.query.groupId]) {
+          this.$router.push({ path: '/dashboard' })
+        } else {
+          await this.accept()
+          return
+        }
       }
+      const state = await sbp('state/latestContractState', this.$route.query.groupId)
+      const invite = state.invites[this.$route.query.secret]
+      if (!invite) {
+        this.ephemeral.errorMsg = L('This invite is not valid.')
+        return this.setStatus('INVALID')
+      } else if (invite && invite.status === 'used') {
```

Comment:
Here especially we want to be checking the status against a constant that's imported from `group.js` or some new file, like `INVITE_USED`. Besides catching typos, this will make it easier to do project-wide refactoring in case any status types need to be redefined.